### ISSUE TITLE
Issue during training linalg.solve

### ISSUE THREAD

**Issue by aleflabo** [epic-kitchens/epic-sounds-annotations#9]
Hi!

I'm trying to reproduce the paper's results but I'm having problems running the training file. 

I've followed the instructions about the environment, datasets and checkpoints. 
It seems there is a problem with the version of torch and relatively the torch.solve function. 
This is the error without changing the code:

`Original Traceback (most recent call last):
  File "/media/hdd/miniconda3/envs/epic/lib/python3.9/site-packages/torch/utils/data/_utils/worker.py", line 302, in _worker_loop
    data = fetcher.fetch(index)
  File "/media/hdd/miniconda3/envs/epic/lib/python3.9/site-packages/torch/utils/data/_utils/fetch.py", line 58, in fetch
    data = [self.dataset[idx] for idx in possibly_batched_index]
  File "/media/hdd/miniconda3/envs/epic/lib/python3.9/site-packages/torch/utils/data/_utils/fetch.py", line 58, in <listcomp>
    data = [self.dataset[idx] for idx in possibly_batched_index]
  File "/home/aleflabo/epic-kitchens/epic-sounds-annotations/src/slowfast/datasets/epicsounds.py", line 119, in __getitem__
    spectrogram = combined_transforms(spectrogram)
  File "/home/aleflabo/epic-kitchens/epic-sounds-annotations/src/slowfast/datasets/spec_augment.py", line 69, in combined_transforms
    return time_mask(freq_mask(time_warp(spec), num_masks=2), num_masks=2)
  File "/home/aleflabo/epic-kitchens/epic-sounds-annotations/src/slowfast/datasets/spec_augment.py", line 23, in time_warp
    warped_spectro, dense_flows = sparse_image_warp(spec, src_pts, dest_pts)
  File "/home/aleflabo/epic-kitchens/epic-sounds-annotations/src/spec_augment/exp/nb_SparseImageWarp.py", line 33, in sparse_image_warp
    flattened_flows = interpolate_spline(
  File "/home/aleflabo/epic-kitchens/epic-sounds-annotations/src/spec_augment/exp/nb_SparseImageWarp.py", line 67, in interpolate_spline
    w, v = solve_interpolation(train_points, train_values, order, regularization_weight)
  File "/home/aleflabo/epic-kitchens/epic-sounds-annotations/src/spec_augment/exp/nb_SparseImageWarp.py", line 113, in solve_interpolation
    X, LU = torch.solve(rhs, lhs)
  File "/media/hdd/miniconda3/envs/epic/lib/python3.9/site-packages/torch/_linalg_utils.py", line 109, in solve
    raise RuntimeError(
RuntimeError: This function was deprecated since version 1.9 and is now removed. Please use the `torch.linalg.solve` function instead.`

This is the error I got when I modify from torch.solve to torch.linalg.solve:

`Original Traceback (most recent call last):
  File "/media/hdd/miniconda3/envs/epic/lib/python3.9/site-packages/torch/utils/data/_utils/worker.py", line 302, in _worker_loop
    data = fetcher.fetch(index)
  File "/media/hdd/miniconda3/envs/epic/lib/python3.9/site-packages/torch/utils/data/_utils/fetch.py", line 58, in fetch
    data = [self.dataset[idx] for idx in possibly_batched_index]
  File "/media/hdd/miniconda3/envs/epic/lib/python3.9/site-packages/torch/utils/data/_utils/fetch.py", line 58, in <listcomp>
    data = [self.dataset[idx] for idx in possibly_batched_index]
  File "/home/aleflabo/epic-kitchens/epic-sounds-annotations/src/slowfast/datasets/epicsounds.py", line 119, in __getitem__
    spectrogram = combined_transforms(spectrogram)
  File "/home/aleflabo/epic-kitchens/epic-sounds-annotations/src/slowfast/datasets/spec_augment.py", line 69, in combined_transforms
    return time_mask(freq_mask(time_warp(spec), num_masks=2), num_masks=2)
  File "/home/aleflabo/epic-kitchens/epic-sounds-annotations/src/slowfast/datasets/spec_augment.py", line 23, in time_warp
    warped_spectro, dense_flows = sparse_image_warp(spec, src_pts, dest_pts)
  File "/home/aleflabo/epic-kitchens/epic-sounds-annotations/src/spec_augment/exp/nb_SparseImageWarp.py", line 33, in sparse_image_warp
    flattened_flows = interpolate_spline(
  File "/home/aleflabo/epic-kitchens/epic-sounds-annotations/src/spec_augment/exp/nb_SparseImageWarp.py", line 67, in interpolate_spline
    w, v = solve_interpolation(train_points, train_values, order, regularization_weight)
  File "/home/aleflabo/epic-kitchens/epic-sounds-annotations/src/spec_augment/exp/nb_SparseImageWarp.py", line 114, in solve_interpolation
    X, LU = torch.linalg.solve(rhs.squeeze(0), lhs.squeeze(0))
RuntimeError: linalg.solve: A must be batches of square matrices, but they are 4 by 2 matrices`

**Comment by JacobChalk** [comment]
Hi!

This is an issue with the [SpecAugment with PyTorch](https://github.com/zcaceres/spec_augment) library using the deprecated `torch.solve` function. In order to use the `torch.linalg.solve` function you need to reverse the arguments, i.e. line 113: `X, LU = torch.solve(rhs, lhs)` should instead become `X = torch.linalg.solve(lhs, rhs)`. This was a very old error which we had ourselves that I forgot existed, we will look at a way to make this clearer/fix this in the repo, thank you for bringing it to our attention!

TLDR: Change `spec_augment/exp/nb_SparseImageWarp.py line 113` to `X = torch.linalg.solve(lhs, rhs)`

UPDATE: We have now pushed a change that contains the fixed version of spec_augment in the EPIC-Sounds repo itself, fixing it as a dependency issue.

**Comment by aleflabo** [comment]
Thank you! That solved my problem
